src/csm_testing/tests/iuf_stages/__main__.py

```diff
@@ -0,0 +1,416 @@
+            sys.exit(1)
+        else:
+            print("TEST CASE: Configmap updated with cfs configurations data")
+            test_cases += 1
+
+        cfs_command = "cray cfs configurations list | grep 'config-minimal-management-dummy-1.0.0'"
+        cfs_output, _ = run_command(cfs_command)
+        if not cfs_output:
+            print("ERROR: No configuration found in cfs")
+            print(f"INFO: Total test cases executed for stage operations: {test_cases}")
+            sys.exit(1)
+        print(f"TEST CASE: CFS configuration found {cfs_output}")
+        test_cases +=1
+    except subprocess.CalledProcessError as e:
+        print(f"Error: {e}")
+        print(f"INFO: Total test cases executed for stage operations: {test_cases}")
+        sys.exit(1)
+
+def prepare_images(ACTIVITY_NAME):
+    global test_cases
+    try:
+        command = f"iuf -a {ACTIVITY_NAME} -m {MEDIA_DIR} run -rv {MEDIA_DIR}/product_vars.yaml -bm {MEDIA_DIR}/management-bootprep.yaml -r prepare-images"
+        run_command(command)
+
+        kubectl_command = (
+            f"kubectl get configmaps -n argo {ACTIVITY_NAME} -o jsonpath="
+            "'{.data.iuf_activity}' | jq '.operation_outputs.stage_params"
+            '["prepare-images"]["prepare-management-images"]'
+            '["sat-bootprep-run"].script_stdout' + "' | xargs -0 echo -e"
+        )
+        kubectl_output, _ = run_command(kubectl_command)
+
+        if not kubectl_output:
+            print("Error: The output for configmap entry of images is null or empty.")
+            print(f"INFO: Total test cases executed for stage operations: {test_cases}")
+            sys.exit(1)
+        else:
+            print("INFO: Found the cm entry...")
+            print(kubectl_output)
+            test_cases += 1
+
+        images_info_cmd = f"""
+        kubectl get configmaps -n argo {ACTIVITY_NAME} -o jsonpath='{{.data.iuf_activity}}' \
+        | jq -r '.operation_outputs.stage_params["prepare-images"]["prepare-management-images"]["sat-bootprep-run"].script_stdout | fromjson | .images[] | "\(.name):\(.final_image_id)"'
+        """ 
+
+        images_name_and_ids, _ = run_command(images_info_cmd)
+
+        if images_name_and_ids:
+            image_lines = images_name_and_ids.splitlines()
+
+            print("TEST CASE: Checking ims images...")
+            for image in image_lines:
+                name, final_image_id = image.split(':')
+                
+                print(f"INFO: Checking cray ims images for image Name: {name.strip()} with image ID: {final_image_id.strip()}")
+                check_ims_cmd=f"cray ims images describe {final_image_id.strip()}"
+                ims_info, _ = run_command(check_ims_cmd)
+                if not ims_info:
+                        print(f"ERROR: Could not find ims image for {name}")
+                        print(f"INFO: Total test cases executed for stage operations: {test_cases}")
+                        sys.exit(1)
```

Comment:
These lines are indented too far. It won't cause a syntax error, but ideally the indenting convention should remain consistent.

```suggestion
               print(f"ERROR: Could not find ims image for {name}")
               print(f"INFO: Total test cases executed for stage operations: {test_cases}")
               sys.exit(1)
```